src/OrchardCore.Modules/OrchardCore.Sitemaps/Controllers/SitemapController.cs

```diff
@@ -65,7 +65,7 @@ public async Task<IActionResult> Index(CancellationToken cancellationToken, stri
             }
             else
             {
-                var work = await Workers.GetOrAdd(sitemap.Path, x => new Lazy<Task<Stream>>(async () =>
+                var work = await Workers.GetOrAdd(sitemapId, x => new Lazy<Task<Stream>>(async () =>
```

Comment:
You also need to use `sitemapId` line 106

Note: There is a bad formatting line 95

Reply:
Done. Good spotting :)